src/api/tokens/tokens.ts

```diff
@@ -0,0 +1,280 @@
+  return tokensByChain[chainId]?.byAddress[address.toLowerCase()];
+}
+
+export function getTokensForChain(chainId: ApiChain): Record<string, TokenEntity> {
+  return tokensByChain[chainId]?.byAddress || {};
+}
+
+export function getAllTokensByChain(): Partial<Record<ApiChain, ChainTokens>> {
+  return tokensByChain;
+}
+
+export function getTokenNative(chainId: ApiChain): TokenNative {
+  const native = getTokenById('NATIVE', chainId);
+  if (!native || !isTokenNative(native)) {
+    throw new Error(`No native token found for chain ${chainId}`);
+  }
+
+  return native;
+}
+
+export function getTokenWrappedNative(chainId: ApiChain): TokenErc20 {
+  const wnative = getTokenById('WNATIVE', chainId);
+  if (!wnative || !isTokenErc20(wnative)) {
+    throw new Error(`No wnative token found for chain ${chainId}`);
+  }
+
+  return wnative;
+}
+
+export function wrappedToNative(token: TokenEntity): TokenEntity {
+  const wnative = getTokenWrappedNative(token.chainId);
+
+  if (areTokensEqual(token, wnative)) {
+    return getTokenNative(token.chainId);
+  }
+
+  return token;
+}
+
+export function nativeToWrapped(token: TokenEntity): TokenErc20 {
+  if (isTokenNative(token)) {
+    return getTokenWrappedNative(token.chainId);
+  }
+
+  return token;
+}
+
+export function isTokenNative(token: TokenEntity): token is TokenNative {
+  return token.type === 'native';
+}
+
+export function isTokenErc20(token: TokenEntity): token is TokenErc20 {
+  return token.type === 'erc20';
+}
+
+export function areTokensEqual(tokenA: TokenEntity, tokenB: TokenEntity): boolean {
+  return (
+    tokenA.chainId === tokenB.chainId &&
+    tokenA.address === tokenB.address &&
+    tokenA.type === tokenB.type
+  );
+}
+
+async function fetchVaultTokensForChain(chainId: ApiChain): Promise<TokenEntity[]> {
+  const vaults = getSingleChainVaults(chainId) || [];
+
+  return vaults.reduce((tokens: TokenEntity[], vault) => {
+    // Native comes from address book
+    if (vault.tokenAddress && vault.tokenAddress !== 'native') {
+      tokens.push({
+        type: 'erc20',
+        id: vault.token,
+        symbol: vault.token,
+        name: vault.token,
+        chainId,
+        oracle: vault.oracle,
+        oracleId: vault.oracleId,
+        address: vault.tokenAddress,
+        decimals: vault.tokenDecimals,
+      });
+    }
+
+    // Skip natives and mooTokens
+    if (
+      vault.earnedTokenAddress &&
+      vault.earnedTokenAddress !== 'native' &&
+      vault.earnedTokenAddress !== vault.earnContractAddress
+    ) {
+      tokens.push({
+        type: 'erc20',
+        id: vault.earnedToken,
+        symbol: vault.earnedToken,
+        name: vault.earnedToken,
+        chainId,
+        oracle: 'tokens', // ???
+        oracleId: vault.earnedOracleId || vault.earnedToken,
+        address: vault.earnedTokenAddress,
+        decimals: vault.earnedTokenDecimals || 18,
+      });
+    }
```

Comment:
Is this reacheable? I don't believe we have any of these

Reply:
just checked and we don't have a mention of 'native' in vault jsons

Reply:
This would pick up gov/earnings vaults (if vault service didn't pre-filter them).

Reply:
even if we had gov vaults there, we don't reference the earnedTokenAddress as native